core/tests/protractor_utils/AdminPage.js

```diff
@@ -207,10 +214,15 @@ var AdminPage = function() {
   };
 
   this._stopOneOffJob = async function(jobName, i) {
-    var text = await (await unfinishedOneOffJobRows.get(i)).getText();
+    await waitFor.visibilityOf(unfinishedOneOffJobRows.get(i));
+    var text = await unfinishedOneOffJobRows.get(i).getText();
     if (text.toLowerCase().startsWith(jobName.toLowerCase())) {
-      await (await unfinishedOneOffJobRows.get(i)).element(
-        by.css('.protractor-test-one-off-jobs-stop-btn')).click();
+      var unfinishedOffJobRowsButton = (
+        await unfinishedOneOffJobRows.get(i)
+      ).element(
+        by.css('.protractor-test-one-off-jobs-stop-btn'));
+      await action.click(
+        'UnfinishedOffJobRowsButton', unfinishedOffJobRowsButton);
     } else {
       await this._stopOneOffJob(jobName, ++i);
     }
```

Comment:
in  `this.expectNumberOfRunningOneOffJobs`, if `count != 0`, we can wait for the first element before counting them